### PR TITLE
gpt-auto, remount-fs: parse fstab= kernel command line option

### PR DESCRIPTION
**Pull request by yuwata**

### CODE REVIEW

File: src/gpt-auto-generator/gpt-auto-generator.c [hunk 1]
```diff
@@ -895,8 +895,16 @@ static int parse_proc_cmdline_item(const char *key, const char *value, void *dat
 
         assert(key);
 
-        if (proc_cmdline_key_streq(key, "systemd.gpt_auto") ||
-            proc_cmdline_key_streq(key, "rd.systemd.gpt_auto")) {
+        if (STR_IN_SET(key, "fstab", "rd.fstab")) {
```
Comment (YHNdnzj) [inline src/gpt-auto-generator/gpt-auto-generator.c]: Hmm, I'm not sure about the duplicate code, especially when the actual performance gain is negligible IMO.

File: src/shared/fstab-util.c [hunk 2]
```diff
@@ -13,15 +13,58 @@
...
+        int r;
+
+        /* If 'enabled' is non-negative, then update the cache with it. */
+        if (enabled >= 0)
+                cached = enabled;
+
+        if (cached >= 0)
+                return cached;
+
+        r = proc_cmdline_parse(parse_proc_cmdline_item, NULL, PROC_CMDLINE_STRIP_RD_PREFIX);
```
Comment (YHNdnzj) [inline src/shared/fstab-util.c]:
```suggestion
        r = proc_cmdline_parse(parse_proc_cmdline_item, &cached, PROC_CMDLINE_STRIP_RD_PREFIX);
```

File: src/gpt-auto-generator/gpt-auto-generator.c [hunk 3]
```diff
@@ -895,8 +895,16 @@ static int parse_proc_cmdline_item(const char *key, const char *value, void *dat
...
+
+                r = value ? parse_boolean(value) : 1;
+                if (r < 0)
+                        log_warning("Failed to parse fstab switch %s. Ignoring.", value);
```
Comment (YHNdnzj) [inline src/gpt-auto-generator/gpt-auto-generator.c]: If you'd still like to keep it, then please log the errno here, even when not used in log message.

File: src/shared/fstab-util.c [hunk 4]
```diff
@@ -13,15 +13,58 @@
...
+        bool *enabled = ASSERT_PTR(data);
+        int r;
+
+        assert(key);
+
+        if (streq(key, "fstab")) {
+
+                r = value ? parse_boolean(value) : 1;
+                if (r < 0)
+                        log_debug("Failed to parse fstab switch %s. Ignoring.", value);
```
Comment (YHNdnzj) [inline src/shared/fstab-util.c]: errno also here

File: src/shared/fstab-util.c (hunk 2 above)
Comment (yuwata) [reply src/shared/fstab-util.c]: Ouch.

File: src/gpt-auto-generator/gpt-auto-generator.c (hunk 1 above)
Comment (yuwata) [reply src/gpt-auto-generator/gpt-auto-generator.c]: okay, I will drop this part.

**Review by YHNdnzj — APPROVED** [review]

File: src/shared/fstab-util.c [hunk 5]
```diff
@@ -13,15 +13,58 @@
...
+        if (streq(key, "fstab")) {
+
+                r = value ? parse_boolean(value) : 1;
+                if (r < 0)
+                        log_debug_errno(r, "Failed to parse fstab switch '%s', ignoring: %m", value);
+                else
+                        *enabled = r;
+        }
+
+        return 0;
```
Comment (poettering) [inline src/shared/fstab-util.c]: proc_cmdline_get_bool() instead?

**Comment by yuwata** [comment]
@YHNdnzj and @poettering Thank you for your review. The suggestion by @poettering is addressed. PTAL.

File: src/basic/proc-cmdline.c [hunk 6]
```diff
@@ -392,13 +392,13 @@ int proc_cmdline_get_key(const char *key, ProcCmdlineFlags flags, char **ret_val
         return cmdline_get_key(args, key, flags, ret_value);
 }
 
-int proc_cmdline_get_bool(const char *key, bool *ret) {
+int proc_cmdline_get_bool(const char *key, ProcCmdlineFlags flags, bool *ret) {
```
Comment (YHNdnzj) [inline src/basic/proc-cmdline.c:395]: `battery-check` can make use of this too. But that's out of the scope for this PR so can be done later.

**Review by YHNdnzj — APPROVED** [review]
LGTM. Maybe the first 2 commits should be backported?

**Comment by keszybz** [comment]
I don't think we should backport this. It is a change in functionality that is generally sensible, but it might cause an issue for people who came to rely on current behaviour. AFAIK, the change doesn't fix any issue reported by users.

**Pull request merged**